File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        A = {}
        Sol = [0, 0]

        for i in range(len(nums)):
            if (target - nums[i]) in A:
                Sol[1] = i
                Sol[0] = A[target - nums[i]]
                return Sol
            else:
                A[nums[i]] = i
        
        return Sol

File: Data Structures & Algorithms/binary-search/submission-2.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0;
        int r = nums.size() - 1;

        while (l <= r) {
            int m = l + ((r - l) / 2);

            if (nums.at(m) == target) {
                return m;
            } else if (target > nums.at(m)) {
                l = m + 1;
            } else if (target < nums.at(m)) {
                r = m - 1;
            }
        }

        return -1;
    }
};
